addon-test-support/scroll.js

```diff
@@ -0,0 +1,47 @@
+import getElementWithAssert from './-private/get-element-with-assert';
+import wait from 'ember-test-helpers/wait';
+
+
+/*
+  Triggers a paint (and therefore flushes any queued up scroll events).
+
+  @method triggerFlushWithPromise
+  @return {RSVP.Promise}
+  @private
+*/
+function triggerFlushWithPromise(){
+  return wait().then(() => {
```

Comment:
Yeah I believe it, but like I said I don't think this is specified behavior necessarily. Basically we're relying on the browser's queue's the line up just right, and I'm not sure that will always be the case (you can see [an example of this here](https://jakearchibald.com/2015/tasks-microtasks-queues-and-schedules/), although they don't go into UI Events and I can't seem to find where those are scheduled in the spec).

I think a better strategy may be to use [dispatchEvent](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/dispatchEvent) to synchronously send the scroll event just after scrollTop/scrollLeft is modified.